cgi/auth.pl

```diff
@@ -47,7 +47,14 @@
 	$status = 200;
 }
 
-print header ( -status => $status );
+print header(-status => $status);
+
+# We need to send the header Access-Control-Allow-Credentials=true so that websites
+# such has hunger.openfoodfacts.org that send a query to world.openfoodfacts.org/cgi/auth.pl
+# can read the resulting response.
+
+print header(-Access_Control_Allow_Credentials => "true");
```

Comment:
As we are sending the header from here, couldn't we first check for referer and limit this to:

`defined env["HTTP_REFERER"] and env["HTTP_REFERER"] =~ /^(?:https?://)\w+$main_domain/`

assuming $main_domain contains openfoodfacts.org/net/localhost

Reply:
Good idea, I added a change like this. The browser will send an Origin header.